realsense_ros_camera/src/base_realsense_node.cpp

```diff
@@ -88,7 +88,13 @@ void BaseRealSenseNode::publishTopics()
     setupPublishers();
     setupStreams();
     if (_enable_tf) {
-        publishStaticTransforms();
+        if (_enable_tf_dynamic) {
+            _transform_thread =
+                boost::shared_ptr<boost::thread>(new boost::thread (boost::bind(&BaseRealSenseNode::prepareTransforms, this)));
```

Comment:
There should be a call to _transform_thread->join() somewhere in the shutdown/destructor sequence.

Reply:
is there a reason why we are using a thread pointer instead of stack allocated thread object? Also any reason why we are using boost threads instead of thread from standard library?

Reply:
Remove thread and publish tf right before we publish image data